contracts/eosiolib/datastream.hpp

```diff
@@ -254,6 +254,13 @@ DataStream& operator >> ( DataStream& ds, std::array<T,N>& v ) {
    return ds;
 }
 
+template<typename DataStream>
```

Comment:
Could we use std::enable_if_t<std::is_fundamental<T>::value == true, int> = 0 and size_of(T) to make this optimization be for all primitives?

Reply:
https://github.com/EOSIO/eos/blob/c6311f255c56824149af203362d8624fc5ce934f/contracts/eosiolib/datastream.hpp#L365-L377

We already do

Reply:
I meant for vector<T>, to optimize more than just vector<T>